git/cmd.py

```diff
@@ -66,10 +66,10 @@
     "with_extended_output",
     "with_exceptions",
     "as_process",
-    "stdout_as_string",
     "output_stream",
-    "with_stdout",
+    "stdout_as_string",
     "kill_after_timeout",
+    "with_stdout",
```

Comment:
This changes only the order in which they are given, not the contents, and the set is equal to the same value as before.

However, I am wondering if `command` should actually be added here. Although the `@overload`-decorated stubs for `Git.execute` define some parameters as keyword-only, in the actual definition no parameter is keyword-only and there is definitional symmetry between `command` and the others. Should I worry about what happens if someone has a `git-command` script that they can usually run as `git command` and tries to run `g.command()` where `g` is a `git.cmd.Git` instance? Or related situations?

Another ramification of the parameters not being keyword-only is that, because they can be passed positionally, i is a breaking change to add a new one elsewhere than the end. Still, regarding https://github.com/gitpython-developers/GitPython/issues/1686#issuecomment-1743680758, if you think a `stdin` parameter should be added as a synonym of `istream`, this could still be done even with `stdin` at the end, with the docstring items documenting the parameters referring to each other to overcome any confusion. I am inclined to say the added complexity is not worthwhile (for example, the function would have to handle the case where they are both passed and with inconsistent values). But a possible argument against this is that the `text` synonym of `universal_newlines` could also be added.

Reply:
> Should I worry about what happens if someone has a `git-command` script that they can usually run as `git command` and tries to run `g.command()` where `g` is a `git.cmd.Git` instance? Or related situations?

In these situations it would be great to know why `command` was put there in the first place, and I for one do not remember. I know there never was an issue related to `command` specifically, which seems to indicate it's a limitation worth ignoring or fixing in a non-breaking fashion (which seems possible).

Regarding `istream`, I'd think maintaining stability would let me sleep better at night especially since you seem to agree that it's used consistently here and in `gitdb`. Probably along with the documentation improvements, all is well and better than it was before, which even in the past didn't seem to have caused enough confusion to cause an issue to be opened.

Reply:
> In these situations it would be great to know why `command` was put there in the first place, and I for one do not remember.

Do you mean why it was *omitted* from the `execute_kwargs` set? If so, my guess is that it was intended to be used positionally while the others were intended to be keyword-only. This is judging by how they are given as keyword-only arguments in the `@overload`s, where they are preceded by a `*,` item in the list of formal parameters, which is absent in the actual function definition (thus causing them to accept both positional and keyword arguments).

But it may be that I am misunderstand what you mean here. It is also possible that their keyword-only status in the `@overload`s was intentionally different from the ultimate definition and intended to provide guidance. (I'm not sure. The `@overload`s are missing most of the arguments, which confuses tooling and seems unintentional.)

Reply:
I am happy to adopt your conclusion in that this is not intentional and since there are negative side-effects, like tooling not working correctly, it's certainly something that could be improved.
If against all odds such an action does create downstream breakage, it could also be undone with a patch release.

Reply:
I don't think we should make the parameters in the real definition actually keyword-only, since that is a breaking change. Or, at least, I would not rush to that. It seems likely to me that, at least for the first few, people are passing them positionally.

I believe it is merely the absence of many parameters from the `@overload`-decorated stubs (which precede the real definition) that is causing VS Code not to show or autocomplete some arguments. Adding the missing parameters to the `@overload`-decorated stubs should be sufficient to fix that, and it shouldn't be a breaking change because *(a)* they are just type stubs, so it doesn't break runtime behavior, *(b)* the actual change seems compatible even relative to what the type stubs said before, and *(c)* GitPython doesn't have static typing stability currently anyway, because although `py.typed` is included in the package, `mypy` checks don't pass (nor, per #1659, do `pyright` checks pass).